state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -811,32 +811,35 @@ pub mod pallet {
 				match current_epoch.saturating_sub(call.original_epoch) {
 					// The call is stale, clean up storage.
 					n if n >= 2 => {
-						T::Broadcaster::clean_up_broadcast_storage(call.broadcast_id);
+						T::Broadcaster::expire_broadcast(call.broadcast_id);
 						Self::deposit_event(Event::<T, I>::FailedForeignChainCallExpired {
 							broadcast_id: call.broadcast_id,
 						});
 					},
-					// Previous epoch, signature is invalid. Re-sign and store.
-					1 => {
-						if let Some(threshold_signature_id) =
-							T::Broadcaster::threshold_resign(call.broadcast_id)
-						{
+					// Previous epoch, signature is invalid. Re-sign but don't broadcast.
+					1 => match T::Broadcaster::re_sign_broadcast(call.broadcast_id, false) {
+						Ok(threshold_signature_id) => {
 							Self::deposit_event(Event::<T, I>::FailedForeignChainCallResigned {
 								broadcast_id: call.broadcast_id,
 								threshold_signature_id,
 							});
 							FailedForeignChainCalls::<T, I>::append(current_epoch, call);
-						} else {
-							// We are here if the Call needs to be resigned, yet no API call data is
-							// available to use. In this situation, there's nothing else that can be
-							// done.
-							log::error!("Foreign Chain Call message cannot be re-signed: Call data unavailable. Broadcast Id: {:?}", call.broadcast_id);
-						}
+						},
+						Err(e) => {
+							// This can happen if a broadcast is still pending despite
+							// since the previous epoch.
+							// TODO: make sure this can't happen.
```

Comment:
I guess in this case, we have to again re-sign and broadcast, right? Alternative, we could emit an event here and all the data in storage and kick off the process again manually, since this seems to be an edge case.

Reply:
Yes - for now we have the option of giving it a bump using governance but of course this isn't ideal. Needs some thought..

Reply:
I have been thinking we could move this auto-resigning of old txs into the broadcast pallet instead, since the broadcast pallet is more aware of the pending/aborted/succeeded states of the txs.